src/test/matching.test.js

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import Enzyme, { mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+// setTestRun activates the Firebase TestDB. It needs to be the first of all relative imports.
+import '../data/firebase/setTestRun';
+import MyPins from '../components/MyPins';
+import { createPin } from '../business/Pin';
+import { createPost } from '../business/Post';
+import { deleteTestDbOnRootLevel, haltIfLiveDB } from './testHelpers';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+const authUser123 = {
+  uid: '123',
+  displayName: 'Max Muster',
+  email: 'maxmuster@gmail.com',
+  photoURL: null,
+};
+const authUser573 = {
+  uid: '573',
+  displayName: 'Nicole Master',
+  email: 'nicolemaster@gmail.com',
+  photoURL: null,
+};
+const pinInfoCategory02 = {
+  userId: '123',
+  title: 'Pin_1',
+  area: {
+    radius: 5,
+    location: {
+      latitude: parseFloat(47.23563352505248),
+      longitude: parseFloat(8.845367431640627),
+    },
+  },
+  categories: {
+    0: true,
+    2: true,
+  },
+};
+const pinInfoCategory1 = {
+  userId: '123',
+  title: 'Pin_2',
+  area: {
+    radius: 5,
+    location: {
+      latitude: parseFloat(47.23563352505211),
+      longitude: parseFloat(8.845367431395730),
+    },
+  },
+  categories: {
+    1: true,
+  },
+};
+const pinInfoCategory0123 = {
+  userId: '573',
+  title: 'Pin_1',
+  area: {
+    radius: 5,
+    location: {
+      latitude: parseFloat(47.23563352505248),
+      longitude: parseFloat(8.845367431640627),
+    },
+  },
+  categories: {
+    0: true,
+    1: true,
+    2: true,
+    3: true,
+  },
+};
+const postInfoCategory1 = {
+  userId: authUser123.uid,
+  title: 'Farmer Market',
+  location: {
+    latitude: parseFloat(47.23563352505248),
+    longitude: parseFloat(8.845367431640627),
+  },
+  category: '1',
+};
+const postInfoCategory2 = {
+  userId: authUser123.uid,
+  title: 'Pub Tour',
+  location: {
+    latitude: parseFloat(47.23563352505211),
+    longitude: parseFloat(8.845367431395730),
+  },
+  category: '2',
+};
+const postInfoCategory0 = {
+  userId: authUser573.uid,
+  title: 'Bike-Tour',
+  location: {
+    latitude: parseFloat(47.23563352505248),
+    longitude: parseFloat(8.845367431640627),
+  },
+  category: '0',
+};
+const postInfoShouldntMatch = {
+  userId: authUser573.uid,
+  title: 'Hiking',
+  location: {
+    latitude: parseFloat(67.23563352505248),
+    longitude: parseFloat(9.845367431640627),
+  },
+  category: '0',
+};
+const pinStatus = { ready1: false, ready2: false, ready3: false };
+const postStatus = {
+  ready1: false, ready2: false, ready3: false, ready4: false,
+};
+
+const setUpForUse = () => {
```

Comment:
copy paste code, could be written as a loop

Reply:
as your shitty comments